src/pages/views.py

```diff
@@ -265,15 +252,53 @@ def get_context_data(self, **kwargs):
         periods = list(self.clusive_user.periods.all())
         current_period = self.clusive_user.current_period
         assigned = self.view == 'assigned'
+        is_teacher = self.clusive_user.can_manage_periods
+        is_guest = self.clusive_user.role == 'GU'
+
+        if is_teacher:
+            readings = get_popular_reads_data(self.clusive_user, periods,
+                                              current_period, assigned_only=assigned)
+        else:
+            # Student
+            if not is_guest:
+                if self.view == 'assigned' or self.view == 'popular':
+                    readings = get_popular_reads_data(self.clusive_user, periods,
+                                                      current_period, assigned_only=assigned)
+                else:
+                    # self.view == 'recent'
+                    readings = get_recent_reads_data(self.clusive_user)[:3]
+            # Guest
+            else:
+                if self.view == 'assigned':
+                    # "Clues to Clusive"
+                    readings = {
+                        'view': 'assigned',
+                        'all': Book.get_featured_books()[:3]
+                    }
+                elif self.view == 'popular':
+                    readings = get_popular_reads_data(self.clusive_user, periods,
+                                                      current_period, assigned_only=False)
+                else:
+                    # self.view == 'recent'
+                    readings = get_recent_reads_data(self.clusive_user)[:3]
+
         context.update({
+            'is_teacher': is_teacher,
+            'is_guest': is_guest,
             'current_period': current_period,
-            'period_name': current_period.name,
+            'period_name': current_period.name if current_period else None,
             'query': None,
-            'data': get_popular_reads_data(self.clusive_user, periods, current_period, assigned_only=assigned),
+            'data': readings,
         })
         return context
 
 
+def get_recent_reads_data(clusive_user):
+    return {
+        'view': 'recent',
+        'all': Paradata.latest_for_user(clusive_user)
+    }
+
 def get_popular_reads_data(clusive_user, periods, current_period, assigned_only):
     # Gather data about books that are trending for dashboard view
```

Comment:
The 'assigned' teacher panel used to be "the most popular readings that are also assigned", but based on the messages we need to change this query.  It is now "the 3 most recently assigned readings". Query needs to be changed to reflect this.

Reply:
Okay, there is now a new query, `BookAssignment.recent_assigned()`, that retrieves the most recently assigned readings for the period, ordered by date assigned.

But, I'm not sure if this makes `BookTrend.top_assigned()` obsolete?

Reply:
I think this function actually returns the _least recently_ assigned!

Reply:
Oops.  Fixed.